apps/wallet-mobile/src/wallet-manager/walletManager.ts

```diff
@@ -9,25 +10,105 @@ import {makeWalletEncryptedStorage} from '../yoroi-wallets/storage'
 import {Keychain} from '../yoroi-wallets/storage/Keychain'
 import {rootStorage} from '../yoroi-wallets/storage/rootStorage'
 import {NetworkId, WalletImplementationId} from '../yoroi-wallets/types'
-import {AddressMode, WalletManagerEvent, WalletManagerSubscription, WalletMeta} from './types'
+import {AddressMode, WalletInfos, WalletManagerEvent, WalletManagerSubscription, WalletMeta} from './types'
 import {isWalletMeta, parseWalletMeta} from './validators'
 
+const thirtyFiveSeconds = 35 * 1e3
+
 export class WalletManager {
```

Comment:
refactored a little, but there is no point in using JS Class, for things that require async constructors, init/isInitialized is just bad implementation it should be a factory. wallet manager needs a complete refactor.